geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -3438,16 +3436,16 @@ public boolean getCopyOnRead() {
 
   @Override
   public boolean removeRoot(InternalRegion rootRgn) {
-    synchronized (rootRegions) {
-      String regionName = rootRgn.getName();
-      InternalRegion found = rootRegions.get(regionName);
-      if (found == rootRgn) {
-        InternalRegion previous = rootRegions.remove(regionName);
-        Assert.assertTrue(previous == rootRgn);
-        return true;
-      }
-      return false;
+
+    String regionName = rootRgn.getName();
+    InternalRegion found = rootRegions.get(regionName);
+    if (found == rootRgn) {
```

Comment:
This code no longer looks correct. The synchronization made this thread safe.
Without the sync you need to just call a method on rootRegions once.
I think you just want to do this: return rootRegions.remove(regionName, rootRgn);